internal/session/handle_login.go

```diff
@@ -21,7 +21,7 @@ func (s *Session) handleLogin(ctx context.Context, tag string, cmd *proto.Login,
 		return response.Bad(tag).WithError(ErrAlreadyAuthenticated)
 	}
 
-	state, err := s.backend.GetState(cmd.GetUsername(), cmd.GetPassword())
+	state, err := s.backend.GetState(ctx, cmd.GetUsername(), cmd.GetPassword(), s.sessionID)
```

Comment:
unnecessary context here and in `(*user).newState(ctx context.Context, ...)`

Reply:
Fixed